rfcs/0000-multiple-control-plane-providers.md

```diff
@@ -0,0 +1,270 @@
+
+As an example of the problem of detecting the exact kind of gateway provider and implementing the configuration accordingly, without it being based again on convention or heuristic, consider the lack of distiction between Istio and OpenShift Service Mesh (OSSM) regarding the structure of their defined `GatewayClass` objects. In fact, such distinction perceived at the level of a `Gateway` object may even go against the purpose of abstraction embodied in the definition of the `GatewayClass` kind, as stated in Gateway API's [specification](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/#gatewayclass):
+
+> We expect that one or more `GatewayClasses` will be created by the infrastructure provider for the user. It allows decoupling of which mechanism (e.g. controller) implements the `Gateways` from the user. For instance, an infrastructure provider may create two `GatewayClasses` named `internet` and `private` to reflect `Gateways` that define Internet-facing vs private, internal applications.
+>
+> [...]
+>
+> The user of the classes will not need to know how `internet` and `private` are implemented.
+
+Differences between `GatewayClasses` provided by different implementations such as the [GatewayClass parameters](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/#gatewayclass-parameters) and [GatewayClass controller selection](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/#gatewayclass-controller-selection) are, according to the spec, only "encouraged" or "RECOMMENDED".
+
+# Prior art
+[prior-art]: #prior-art
+
+N/A.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+1. Is having multiple service mesh control planes (i.e. multiple gateway providers) within the same cluster a real use case or rather just a technical possibility that we choose not to support?<br/>
+  **_A:_**<br/><br/>
+2. Does a Kuadrant instance need to be able to target more than one control plane/gateway provider? (See [Rationale and alternatives](#rationale-and-alternative).)<br/>
+  **_A:_**<br/><br/>
+3. Should targeting a control plane/gateway provider be required to be explicit (e.g. in a `Kuadrant` CR) or it can based on convention/heuristic (e.g. default to Istio)? _Note:_ I am personally inclined to it needs to be explicit, hence the proposal in this RFC.<br/>
+  **_A:_**<br/><br/>
+4. What gateway providers should we initially support?<br/>
+  **_A:_** Istio and OpenShift Service Mesh (OSSM).<br/><br/>
+5. What other gateway providers could we aim to support in the future? _Possible answers:_ [Envoy Gateway](https://gateway.envoyproxy.io/v0.2.0/), [Contour](https://projectcontour.io), other API gateways not based on Envoy, ...<br/>
+  **_A:_** Only gateways and perhaps sidecar proxies based on Envoy for quite some time, likely.<br/><br/>
+6. Is "gateway provider" an adequate term? What about supporting injecting Kuadrant into sidecar proxies in the future?<br/>
```

Comment:
Gateway provider seems adequate a sidcar could be thought of as another form of gateway